pkgs/development/interpreters/gnudatalanguage/default.nix

```diff
@@ -0,0 +1,167 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+# Native build inputs for the compiler
+, cmake
+, wrapGAppsHook
+# Build dependencies
+, readline
+, ncurses # alt: curses
+, zlib
+, gsl
+, openmp
+# GraphicsMagick are preferred over ImageMagick
+# https://github.com/gnudatalanguage/gdl/blob/master/CMakeLists.txt#L600-L601
+, graphicsmagick # alt: imagemagick
+, netcdf
+, hdf4
+, hdf5
+, fftw, fftwFloat, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib # a C library for PostScript
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg # for hdf4 and hdf5
+, enableLibtirpc ? stdenv.isLinux, libtirpc
+, pythonInterpreter ? python3, python3
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+# Choose mpich because of it builds on aarch and darwin
+, libraryMPI ? mpich # alt: openmpi
+, mpich
+# Unfree optional dependency
```

Comment:
Remember to conditionally update `meta.licenses` accordingly.

Reply:
> Remember to conditionally update `meta.licenses` accordingly.

This is not needed IMO, because no matter what is the license of `gnudatalanguage`, with szip enabled it will result in evaluating `szip` which will result in a evaluation failure for `gnudatalanguage` itself as well - with the known error message indicating what line the evaluation failure came from which indicates what package in the dependencies has an unfree license..